erpnext/projects/doctype/task/task.json

```diff
@@ -153,8 +154,10 @@
    "fieldname": "exp_start_date",
    "fieldtype": "Date",
    "label": "Expected Start Date",
+   "mandatory_depends_on": "eval: doc.is_template == 0;",
    "oldfieldname": "exp_start_date",
-   "oldfieldtype": "Date"
+   "oldfieldtype": "Date",
+   "reqd": 1
```

Comment:
Let's not keep the Date fields mandatory. In the report, we can either filter out such tasks, or else we can keep the delay as 0 for the tasks which don't have the end date set.

Reply:
As we are not keeping the end date as mandatory, I have also made the start and end date filters in the report non-mandatory. If we keep the filters as mandatory then tasks with no date get filtered out and won't appear in the report at all.